Apps/W1/Shopify/test/Logs/ShpfySkippedRecordLogTest.Codeunit.al

```diff
@@ -0,0 +1,838 @@
+    var
+        Customer: Record Customer;
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        SkippedRecord: Record "Shpfy Skipped Record";
+        PostedInvoiceExport: Codeunit "Shpfy Posted Invoice Export";
+        LibrarySales: Codeunit "Library - Sales";
+    begin
+        // [SCENARIO] Log skipped record when sales invoice export is skipped because not existing shopify customer.
+        Initialize();
+
+        // [GIVEN] Customer
+        LibrarySales.CreateCustomer(Customer);
+        // [GIVEN] Sales Invoice
+        CreateSalesInvoiceHeader(SalesInvoiceHeader, Customer."No.", '');
+
+        // [WHEN] Invoke Shopify Posted Invoice Export
+        PostedInvoiceExport.ExportPostedSalesInvoiceToShopify(SalesInvoiceHeader);
+
+        // [THEN] Related record is created in shopify skipped record table.
+        SkippedRecord.SetRange("Record ID", SalesInvoiceHeader.RecordId);
+        LibraryAssert.IsTrue(SkippedRecord.FindLast(), 'Skipped record is not created');
+        LibraryAssert.AreEqual('Customer does not exists as Shopify company of customer.', SkippedRecord."Skipped Reason", 'Skipped reason is not as expected');
+    end;
+
+    [Test]
+    procedure UnitTestLogSalesInvoiceWithNotExistingShopifyPaymentTerms()
+    var
+        Customer: Record Customer;
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        SkippedRecord: Record "Shpfy Skipped Record";
+        PostedInvoiceExport: Codeunit "Shpfy Posted Invoice Export";
+        PaymentTermsCode: Code[10];
+    begin
+        // [SCENARIO] Log skipped record when sales invoice export is skipped because of not existing shopify payment terms.
+        Initialize();
+
+        // [GIVEN] Customer
+        Customer := ShpfyInitializeTest.GetDummyCustomer();
+        // [GIVEN] Shopify Customer
+        CreateShopifyCustomer(Customer);
+        // [GIVEN] Payment Terms Code
+        PaymentTermsCode := Any.AlphanumericText(10);
+        // [GIVEN] Sales Invoice
+        CreateSalesInvoiceHeader(SalesInvoiceHeader, Customer."No.", PaymentTermsCode);
+
+        // [WHEN] Invoke Shopify Posted Invoice Export
+        PostedInvoiceExport.ExportPostedSalesInvoiceToShopify(SalesInvoiceHeader);
+
+        // [THEN] Related record is created in shopify skipped record table.
+        SkippedRecord.SetRange("Record ID", SalesInvoiceHeader.RecordId);
+        LibraryAssert.IsTrue(SkippedRecord.FindLast(), 'Skipped record is not created');
+        LibraryAssert.AreEqual(StrSubstNo('Payment terms %1 do not exist in Shopify.', PaymentTermsCode), SkippedRecord."Skipped Reason", 'Skipped reason is not as expected');
+    end;
+
+    [Test]
+    procedure UnitTestLogSalesInvoiceWithCustomerNoIsDefaultCustomerNo()
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        Customer: Record Customer;
+        SkippedRecord: Record "Shpfy Skipped Record";
+        Shop2: Record "Shpfy Shop";
```

Comment:
Why an extra shop? Can it have a better name?

Reply:
Did this to not have to clear Default Customer No and delete Shopify Customer Templates in other tests. But have idea to fix this place.